GetJobInstanceStatus - Does not work

Below Gradle file does not work. 


plugins {
  id "io.saagie.gradle-saagie-dataops-plugin" version "1.1.10"
}

saagie {
    server {
        url = 'https://saagie-beta.prod.saagie.io'
        login = saagieuserid
        password = saagiepassword
        environment = 4
        jwt = true
    }
    project {
        id = 'ec8c5cea-3dfd-4496-87a0-89f69101dccd'
    }
   jobinstance {
          id = '53b75b58-a136-499f-9370-f05abccb2a76'
   }
}






saagies-mbp:Gradle shivakumar$ gradle -b build.gradle.projectGetJobInstanceStatus projectsGetJobInstanceStatus

FAILURE: Build failed with an exception.

* Where:
Build file '/Users/shivakumar/Documents/Demo/Gradle/build.gradle.projectGetJobInstanceStatus' line: 16

* What went wrong:
A problem occurred evaluating root project 'Gradle'.
> Could not find method jobinstance() for arguments [build_gradle_43uqvbeybn5e8ubjy1oo1vg2s$_run_closure1$_closure4@5e002c28] on object of type io.saagie.plugin.dataops.DataOpsExtension.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 462ms

--- a/src/main/groovy/io/saagie/plugin/dataops/DataOpsExtension.groovy
+++ b/src/main/groovy/io/saagie/plugin/dataops/DataOpsExtension.groovy
@@ -17,7 +17,7 @@ class DataOpsExtension {
 
     Job job = new Job()
     JobVersion jobVersion = new JobVersion()
-    JobInstance jobInstance = new JobInstance()
+    JobInstance jobinstance = new JobInstance()
     PipelineInstance pipelineInstance = new PipelineInstance()
 
     Pipeline pipeline = new Pipeline()
@@ -39,8 +39,8 @@ class DataOpsExtension {
         jobVersion.with(closure)
     }
 
-    Object jobInstance(Closure closure) {
-        jobInstance.with(closure)
+    Object jobinstance(Closure closure) {
+        jobinstance.with(closure)
     }
 
     Object pipeline(Closure closure) {

--- a/src/main/groovy/io/saagie/plugin/dataops/clients/SaagieClient.groovy
+++ b/src/main/groovy/io/saagie/plugin/dataops/clients/SaagieClient.groovy
@@ -367,12 +367,12 @@ class SaagieClient {
 
     String getJobInstanceStatus() {
         logger.info('Starting getJobInstanceStatus task')
-        if (configuration?.jobInstance?.id == null        ) {
+        if (configuration?.jobinstance?.id == null        ) {
             logger.error(BAD_PROJECT_CONFIG.replaceAll('%WIKI%', PROJECTS_GET_JOB_INSTANCE_STATUS))
             throw new InvalidUserDataException(BAD_PROJECT_CONFIG.replaceAll('%WIKI%', PROJECTS_GET_JOB_INSTANCE_STATUS))
         }
 
-        logger.debug('Using config [project={}, jobInstance={}]', configuration.project, configuration.jobInstance)
+        logger.debug('Using config [project={}, jobInstance={}]', configuration.project, configuration.jobinstance)
 
         Request projectJobInstanceStatusRequest = saagieUtils.getProjectJobInstanceStatusRequest()
         try {
@@ -548,7 +548,7 @@ class SaagieClient {
 
     String stopJobInstance() {
         logger.info('Starting stopJobInstance task')
-        if (configuration?.jobInstance?.id == null) {
+        if (configuration?.jobinstance?.id == null) {
             logger.error(BAD_PROJECT_CONFIG.replaceAll('%WIKI%', PROJECT_STOP_JOB_INSTANCE_TASK))
             throw new InvalidUserDataException(BAD_PROJECT_CONFIG.replaceAll('%WIKI%', PROJECT_STOP_JOB_INSTANCE_TASK))
         }

--- a/src/main/groovy/io/saagie/plugin/dataops/utils/SaagieUtils.groovy
+++ b/src/main/groovy/io/saagie/plugin/dataops/utils/SaagieUtils.groovy
@@ -343,7 +343,7 @@ class SaagieUtils {
     }
 
     Request getProjectJobInstanceStatusRequest() {
-        JobInstance jobInstance = configuration.jobInstance;
+        JobInstance jobInstance = configuration.jobinstance;
         logger.debug('Generating getProjectJobsRequest [projectId={}]', jobInstance.id)
 
         def jsonGenerator = new JsonGenerator.Options()
@@ -536,7 +536,7 @@ class SaagieUtils {
     }
 
     Request getStopJobInstanceRequest() {
-        JobInstance jobInstance = configuration.jobInstance
+        JobInstance jobInstance = configuration.jobinstance
         logger.debug('Generating getStopJobInstanceRequest for job instance [id={}]', jobInstance.id)
 
         def jsonGenerator = new JsonGenerator.Options()

--- a/src/test/groovy/io/saagie/plugin/dataops/clients/SaagieClientTest.groovy
+++ b/src/test/groovy/io/saagie/plugin/dataops/clients/SaagieClientTest.groovy
@@ -55,7 +55,7 @@ class SaagieClientTest extends Specification {
                 name = tempFile.absolutePath
             }
         }
-        configuration.jobInstance {
+        configuration.jobinstance {
             id = 'jobInstanceId'
         }
         configuration.pipeline {
@@ -341,7 +341,7 @@ class SaagieClientTest extends Specification {
         enqueueToken()
 
         client = new SaagieClient(configuration)
-        client.configuration.jobInstance.id = null
+        client.configuration.jobinstance.id = null
 
         when:
         String getJobInstanceStatusResult = client.getJobInstanceStatus()

--- a/src/test/groovy/io/saagie/plugin/job/JobGetInstanceStatusTaskTests.groovy
+++ b/src/test/groovy/io/saagie/plugin/job/JobGetInstanceStatusTaskTests.groovy
@@ -68,7 +68,7 @@ class JobGetInstanceStatusTaskTests extends Specification {
                     environment = 2
                 }
 
-                jobInstance {
+                jobinstance {
                     id = "jobInstanceId"
                 }
             }

--- a/src/test/groovy/io/saagie/plugin/job/JobStopTaskTests.groovy
+++ b/src/test/groovy/io/saagie/plugin/job/JobStopTaskTests.groovy
@@ -68,7 +68,7 @@ class JobStopTaskTests extends Specification {
                     environment = 4
                 }
                 
-                jobInstance {
+                jobinstance {
                     id = 'job-instance-id'
                 }
             }
@@ -119,7 +119,7 @@ class JobStopTaskTests extends Specification {
                     environment = 2
                 }
                 
-                jobInstance {
+                jobinstance {
                     id = 'bad-id'
                 }
             }

